### PR TITLE
feat: allow to show/hide navigation items

### LINKED ISSUE #7758: Ability to show/hide navigation items
### Is your enhancement related to a problem? Please describe

As we add more navigation items there are more items that are rarely used, extensions that add 2 items but you only use one of them, or cases where you want to temporarily hide the icon without uninstalling the extension.

### Describe the solution you'd like

Users should be able to right-click on the navigation bar to see a list of all navigation items in a checkbox list, and easily add or remove the visible items.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Allow to toggle the visibility of left navbar items

### Screenshot / video of UI

https://github.com/user-attachments/assets/fbfadd1a-bfb1-475d-9ff8-1232a63ef0e4

### What issues does this PR fix or reference?

fixes https://github.com/containers/podman-desktop/issues/7758

### How to test this PR?

Right click on an item like Containers, Pods and select `Hide <item>`
or right click on any place on the navbar and then check or uncheck the checkbox for a given item

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by deboer-tim — CHANGES_REQUESTED** [review]
Function working great, but I have some nits and ran into related issues:
- The tooltip stays open when the menu appears. This is the same issue seen in #6660 but not sure the solution would be the same, when you open the menu the tooltip should go away.
- I don't think the menu items should be sorted. Although there's a separate Hide option I was looking for the current menu in the list and this made it harder to scan for.
- Items like 'ConfigMaps & Secrets' are missing the ampersand and show up like 'ConfigMaps  Secrets'. It might be interpreted as an accelerator key and need to be escaped.
- When using `yarn watch` the first two menu items are `Copy Link` and `Inspect`. I'm assuming the Inspect would be gone in production build but Copy Link is distracting (breaks the flow and makes me realize this is electron). Is there a quick way to remove it, or maybe we can track via separate issue?

**Comment by benoitf** [comment]
fixed 2 and 3 (label escaping and sorting)

for 1. and 4. it's already there so should be other PRs/issue.

**Comment by deboer-tim** [comment]
> fixed 2 and 3 (label escaping and sorting)

Thanks for fixing!

> for 1. and 4. it's already there so should be other PRs/issue.

Yeah, it's just unfortunate that users probably wouldn't have known we had these issues until we tell them about this feature. I opened #8335 for #1. For #4 I noticed several other native menus so I opened the broader design question to start with in #8336.

**Review by deboer-tim — APPROVED** [review]

**Review by axel7083 — CHANGES_REQUESTED** [review]
Weird behaviour with the account and settings, where we should probably not see the hide action?


https://github.com/user-attachments/assets/23c2ae3c-6ee1-43ae-bdd7-246daf056c9b

If am hidding all kubernetes, the group menu should disapear ?

image

**Comment by benoitf** [comment]
@axel7083 should be fixed

**Review by axel7083 — COMMENTED** [review]
Works nicely

File: packages/renderer/src/AppNavigation.svelte [hunk 1]
```diff
@@ -47,15 +47,18 @@ function clickSettings(b: boolean) {
   {#each $navigationRegistry as navigationRegistryItem}
     <!-- This is a section -->
     {#if navigationRegistryItem.type === 'section' && navigationRegistryItem.enabled}
-      <NavSection tooltip={navigationRegistryItem.name}>
-        <svelte:component this={navigationRegistryItem.icon.iconComponent} size={iconSize} slot="icon" />
+      {@const allItemsHidden = (navigationRegistryItem.items ?? []).every(item => item.hidden)}
+      {#if !allItemsHidden}
```
Comment (axel7083) [inline packages/renderer/src/AppNavigation.svelte:51]: Why declaring a variable which has a single usage ?

Comment (axel7083) [inline packages/renderer/src/AppNavigation.svelte:51]:
```suggestion
      {#if !(navigationRegistryItem.items ?? []).every(item => item.hidden)}
```

**Review by axel7083 — APPROVED** [review]
LGTM

File: packages/renderer/src/AppNavigation.svelte (hunk 1 above)
Comment (benoitf) [reply packages/renderer/src/AppNavigation.svelte:51]: it looks a little bit too long to be directly a condition. I find it more readable.

Comment (axel7083) [reply packages/renderer/src/AppNavigation.svelte:51]: I approved so that's not blocking for me

**Pull request merged**